feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/views/priceCharts/PriceChartsView.kt

```diff
@@ -0,0 +1,174 @@
+
+    init {
+        View.inflate(context, R.layout.view_price_charts, this)
+        controller = ChartController(priceChart, this)
+        setEmptyState()
+    }
+
+    fun setTitle(title: String) {
+        priceChartTitle.text = title
+    }
+
+    fun setCharts(charts: List<PriceChartModel>) {
+        this.charts = charts
+
+        priceChartButtons.removeAllViews()
+        charts.forEachIndexed { index, priceChartModel ->
+            if (index > 0) {
+                priceChartButtons.addView(getSpace())
+            }
+
+            priceChartButtons.addView(getButton(priceChartModel.name, index))
+        }
+
+        if (charts.isEmpty()) {
+            setEmptyState()
+        } else {
+            selectChart(0)
+        }
+    }
+
+    fun setTextInjectors(
+        priceTextInjector: PriceTextInjector,
+        priceChangeTextInjector: PriceChangeTextInjector,
+        dateTextInjector: DateChartTextInjector
+    ) {
+        this.priceTextInjector = priceTextInjector
+        this.priceChangeTextInjector = priceChangeTextInjector
+        this.dateTextInjector = dateTextInjector
+    }
+
+    override fun onSelectEntry(startEntry: Entry, selectedEntry: Entry, isLastEntry: Boolean) {
+        priceTextInjector?.format(selectedEntry.y, priceChartCurrentPrice)
+        priceChangeTextInjector?.format(startEntry.y, selectedEntry.y, priceChartPriceChange)
+        dateTextInjector?.format(selectedEntry.x.roundToLong(), isLastEntry, priceChartDate)
+    }
+
+    private fun setEmptyState() {
+        val periods = 2
+        val entriesCount = 100
+        val entries = List(entriesCount) {
+            val sinY = sin(periods * Math.PI * 2 * it / entriesCount)
+            Entry(it.toFloat(), sinY.toFloat())
+        }
+
+        controller.setEntries(entries)
+
+        showLoadingState(true)
+    }
+
+    private fun showLoadingState(show: Boolean) {
+        priceChartDate.isInvisible = show
+        priceChartPriceChange.isInvisible = show
+        priceChartCurrentPrice.isInvisible = show
+        priceChartPriceChangeShimmering.isVisible = show
+        priceChartCurrentPriceShimmering.isVisible = show
+
+        controller.useNeutralColor(show)
+        controller.showYAxis(!show)
+    }
+
+    private fun selectChart(index: Int) {
+        getButtons().forEachIndexed { i, view ->
+            view.isSelected = i == index
+        }
+
+        val currentChart = charts[index]
+        if (currentChart is PriceChartModel.Loading) {
+            setEmptyState()
+            return
+        } else if (currentChart is PriceChartModel.Chart) {
+            controller.setEntries(currentChart.asEntries())
+
+            showLoadingState(false)
+        }
+    }
+
+    private fun getButtons(): List<TextView> {
+        return priceChartButtons.children
+            .toList()
+            .filterIsInstance<TextView>()
+    }
+
+    private fun getButton(text: String, index: Int): View {
```

Comment:
I suggest to rename it to `createButton` since there is `getButtons` that doesnt create anything